src/AWS/Shared/Storage/DynamoDBStorage.cs

```diff
@@ -118,6 +123,30 @@ private void CreateClient()
                 var credentials = new BasicAWSCredentials(this.accessKey, this.secretKey);
                 this.ddbClient = new AmazonDynamoDBClient(credentials, new AmazonDynamoDBConfig {RegionEndpoint = AWSUtils.GetRegionEndpoint(this.service)});
             }
+            else if (!string.IsNullOrEmpty(this.profileName))
+            {
+                // AWS DynamoDB instance (auth via explicit credentials and token found in a named profile)
+                var chain = new CredentialProfileStoreChain();
+                if (chain.TryGetAWSCredentials(this.profileName, out var credentials))
+                {
+                    var immutableCredentials = credentials.GetCredentials();
+                    var sessionCredentials = new SessionAWSCredentials(
```

Comment:
Why not just pass the credentials retrieved the chain directly? I believe this method will have issues with tokens expiring.

Reply:
`AmazonDynamoDBClient ` constructor does not accept `ImmutableCredentials` and it doesn't look like `AWSCredentials` (which `SessionAWSCredentials` extends) are interchangeable with `ImmutableCredentials`.

Reply:
You should be able to pass `credentials` directly to the `AmazonDynamoDBClient` constructor.

Reply:
Ah yes, you're right, it can be simplified. The type of `credentials` is `AWSCredentials`, so the `AmazonDynamoDBClient` constructor does accept those.

I updated the code, ran it locally just in case and can confirm the `ddbClient` works correctly.

Thanks for the suggestion.